test/e2e/storage/volume_provisioning.go

```diff
@@ -126,8 +127,15 @@ func testDynamicProvisioning(t storageClassTest, client clientset.Interface, cla
 	// - The second one runs grep 'hello world' on /mnt/test.
 	// If both succeed, Kubernetes actually allocated something that is
 	// persistent across pods.
-	By("checking the created volume is writable")
-	runInPodWithVolume(client, claim.Namespace, claim.Name, "echo 'hello world' > /mnt/test/data")
+	By("checking the created volume is writable and has the PV's mount options")
+	command := "echo 'hello world' > /mnt/test/data"
+	// We give the first pod the secondary responsibility of checking the volume has
+	// been mounted with the PV's mount options, if the PV was provisioned with any
+	for _, option := range pv.Spec.MountOptions {
+		// -M to define mountpoint, -o to define output column, -n to exclude header
+		command += fmt.Sprintf(" && [[ \",`findmnt -M /mnt/test/data -o OPTIONS -n`,\" = *\",%s,\"* ]]", option)
```

Comment:
Is there a mount options delimiter you can include in the `findmnt` test? Eg. if a mount option is a subset of another mount option you'll get a false positive.  Eg: `atime` is a subset of `noatime`

Reply:
It probably doesn't matter here but could be more efficient to capture the `fndmnt` options once and then do a bash "includes" string compare for each class option

Reply:
each option is surrounded by commas (we add commas to the beginning and end of findmnt), so it will work because it's checking if the full ",noatime," exists.

yeah I just wanted the simplest/shortest bash expression so settled on this
 https://stackoverflow.com/questions/29300806/a-bash-script-to-check-if-a-string-is-present-in-a-comma-separated-list-of-strin

Reply:
the `-M` option does not appear to be supported in rhel, centos, fedora images. It is supported in ubuntu. It looks like the `--target` findmnt option is more universally supported.